index.js

```diff
@@ -64,82 +83,151 @@ function withFilter(predicate) {
 }
 exports.withFilter = withFilter;
 
-function createWriter(predicate) {
+function withRecords(records, predicate) {
+  var recordMap = {};
+
+  records.forEach(function(RecordType) {
+    var rec = new RecordType({});
+
+    if (!recordName(rec)) {
+      throw new Error('Cannot (de)serialize Record() without a name field');
```

Comment:
I'd drop the word `field` here, as name is distinct from the actual record fields